### PR TITLE
fix(themes): 🐛 Fix memory leak error in product-card

### PR DESCRIPTION
**Pull request by eslamoo**
What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

* 

What is the current behaviour? (You can also link to an open issue here)

* 

What is the new behaviour? (You can also link to the ticket here)

* 

Does this PR introduce a breaking change?

* 

Screenshots (If appropriate)

*

### CODE REVIEW

File: src/assets/js/partials/product-card.js [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+    });
+  }
+
+  if (!isListenerAttached) {
+    salla.wishlist.event.onAdded((event, id) => toggleFavoriteIcon(id));
+    salla.wishlist.event.onRemoved((event, id) => toggleFavoriteIcon(id, false));
+    isListenerAttached = true; // Mark the listener as attached
+  }
+})
+})();
```
Comment (jalmatari) [inline src/assets/js/partials/product-card.js:21]: I think It's better to move this to app.js file

**Review by jalmatari — CHANGES_REQUESTED** [review]

**Comment by SallaDev** [comment]
**Approve:**
- can not merge pr (https://github.com/SallaApp/theme-raed/pull/445) needs at least `one approve ❗️`



Merge by: @jalmatari

**Review by jalmatari — APPROVED** [review]

**Comment by SallaDev** [comment]
https://github.com/SallaApp/theme-raed/pull/445 is been merged ✅

-------------------

Merge by: @jalmatari

**Pull request merged**